acvm/src/lib.rs

```diff
@@ -58,42 +65,38 @@ pub trait PartialWitnessGenerator {
     fn solve(
         &self,
         initial_witness: &mut BTreeMap<Witness, FieldElement>,
-        opcodes: Vec<Opcode>,
+        mut gates_to_resolve: Vec<Opcode>,
     ) -> Result<(), OpcodeResolutionError> {
-        if opcodes.is_empty() {
-            return Ok(());
-        }
-        let mut unsolved_opcodes: Vec<Opcode> = Vec::new();
-
-        for opcode in opcodes.into_iter() {
-            let resolution = match &opcode {
-                Opcode::Arithmetic(expr) => ArithmeticSolver::solve(initial_witness, expr),
-                Opcode::BlackBoxFuncCall(bb_func) => {
-                    Self::solve_black_box_function_call(initial_witness, bb_func)
+        let mut unresolved_gates: Vec<Opcode> = Vec::new();
+        let mut blocker = Blocks::default();
```

Comment:
`blocker` reads like a blocking operation to me. Perhaps we should rename this to `blocks` or similar

Reply:
Done